### PR TITLE
[DRK] Audit abilities

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Some changes from PR #2877 
Audits DRK abilities

NOTE: Core changes to make some of this abilities actually work __not included__

## Steps to test these changes

!changejob drk 99
Use abilities

### CODE REVIEW

File: scripts/globals/effects/soul_enslavement.lua [hunk 1]
```diff
@@ -4,12 +4,16 @@
 local effectObject = {}
 
 effectObject.onEffectGain = function(target, effect)
+    target:addMod(xi.mod.ENSPELL, 22)
```
Comment (WinterSolstice8) [inline scripts/globals/effects/soul_enslavement.lua]: How does this interact with other enspells like endark?

File: scripts/globals/effects/souleater.lua [hunk 2]
```diff
@@ -6,14 +6,21 @@ require("scripts/globals/status")
 local effectObject = {}
 
 effectObject.onEffectGain = function(target, effect)
+    -- Enmity bonus is handled in the CalculateEnmityBonus function inside enmity_container.cpp
+    local subPower = effect:getSubPower()
+
     target:addMod(xi.mod.ACC, 25)
+    target:addMod(xi.mod.ZANSHIN, subPower)
```
Comment (WinterSolstice8) [inline scripts/globals/effects/souleater.lua]: Zanshin? Can't find a source for this but I can't say I know drk very well

File: scripts/globals/job_utils/dark_knight.lua [hunk 3]
```diff
@@ -87,36 +90,49 @@ end
...
 xi.job_utils.dark_knight.useNetherVoid = function(player, target, ability)
-    player:addStatusEffect(xi.effect.NETHER_VOID, 8, 1, 30)
+    local power = 50 + player:getMod(xi.mod.ENHANCES_NETHER_VOID) + 2 * player:getJobPointLevel(xi.jp.NETHER_VOID_EFFECT)
+
+    player:addStatusEffect(xi.effect.NETHER_VOID, power, 0, 60)
 end
 
 xi.job_utils.dark_knight.useScarletDelirium = function(player, target, ability)
-    player:addStatusEffect(xi.effect.SCARLET_DELIRIUM, 8, 1, 90)
+    local jpValue  = player:getJobPointLevel(xi.jp.SCARLET_DLRIUM_DURATION)
```
Comment (WinterSolstice8) [inline scripts/globals/job_utils/dark_knight.lua]: `DLRIUM`, typo?

Comment (Xaver-DaRed) [reply scripts/globals/job_utils/dark_knight.lua]: Nope, although I though it was at first myself.

imagen

I could change it in a later PR though. Out of scope for this one.

File: scripts/globals/effects/souleater.lua (hunk 2 above)
Comment (Xaver-DaRed) [reply scripts/globals/effects/souleater.lua]: It's for this piece of gear

imagen

Comment (WinterSolstice8) [reply scripts/globals/effects/souleater.lua]: oh, guess i didn't look hard enough

File: scripts/globals/job_utils/dark_knight.lua (hunk 3 above)
Comment (WinterSolstice8) [reply scripts/globals/job_utils/dark_knight.lua]: yup save it for later

File: scripts/globals/effects/soul_enslavement.lua (hunk 1 above)
Comment (Xaver-DaRed) [reply scripts/globals/effects/soul_enslavement.lua]: Good question, actually. I don't know.

Time to hit retail and find out. I can leave a TODO in the meantime

Comment (gweivyth) [reply scripts/globals/effects/soul_enslavement.lua]: Not sure if you ever revisited this but I'm leveling DRK tonight so I'll test this out.

**Comment by TeoTwawki** [comment]
- needs conflicts resolved
- enspell question needs addressed

**Comment by Xaver-DaRed** [comment]
Ready for review.
I leveled a DRK to 96 to know which effects get overriden and which dont

**Review by WinterSolstice8 — APPROVED** [review]

**Review by zach2good — APPROVED** [review]

**Pull request merged**